cmd/fleetctl/get_test.go

```diff
@@ -283,74 +282,75 @@ func TestGetHosts(t *testing.T) {
 +------+------------+----------+-----------------+--------+
 `
 
+	jsonPrettify := func(t *testing.T, v string) string {
+		var i interface{}
+		err := json.Unmarshal([]byte(v), &i)
+		require.NoError(t, err)
+		indented, err := json.MarshalIndent(i, "", "  ")
+		require.NoError(t, err)
+		return string(indented)
+	}
+	yamlPrettify := func(t *testing.T, v string) string {
+		var i interface{}
+		err := yaml.Unmarshal([]byte(v), &i)
+		require.NoError(t, err)
+		indented, err := yaml.Marshal(i)
+		require.NoError(t, err)
+		return string(indented)
+	}
 	tests := []struct {
-		name        string
-		goldenFile  string
-		unmarshaler func(data []byte, v interface{}) error
-		scanner     func(s string) []string
-		args        []string
+		name       string
+		goldenFile string
+		scanner    func(s string) []string
+		prettifier func(t *testing.T, v string) string
+		args       []string
 	}{
 		{
-			name:        "get hosts --json",
-			goldenFile:  "expectedListHostsJson.json",
-			unmarshaler: json.Unmarshal,
+			name:       "get hosts --json",
+			goldenFile: "expectedListHostsJson.json",
 			scanner: func(s string) []string {
-				var parts []string
-				scanner := bufio.NewScanner(bytes.NewBufferString(s))
-				for scanner.Scan() {
-					parts = append(parts, scanner.Text())
-				}
-				return parts
+				parts := strings.Split(s, "}\n{")
+				return []string{parts[0] + "}", "{" + parts[1]}
```

Comment:
Why not amend the golden JSON?

Reply:
Not sure if I understand how you are suggesting to amend the json file, but this is because we output a concatenation of JSONs which is not a valid JSON. Not particularly happy about how this is just yet, but it feels like an improvement. Any better ideas?

Reply:
Nevermind, I see the issue now.